experimental/material_3_demo/lib/main.dart

```diff
@@ -0,0 +1,724 @@
+          icon: onLightMode
+              ? const Icon(Icons.wb_sunny_outlined)
+              : const Icon(Icons.wb_sunny),
+          onPressed: handleModeChange,
+        ),
+        IconButton(
+            onPressed: handleVersionChange,
+            icon: onMaterial3
+                ? const Icon(Icons.filter_3)
+                : const Icon(Icons.filter_2)),
+        PopupMenuButton(
+          icon: const Icon(Icons.more_vert),
+          shape:
+              RoundedRectangleBorder(borderRadius: BorderRadius.circular(10)),
+          itemBuilder: (context) {
+            return List.generate(colorOptions.length, (index) {
+              return PopupMenuItem(
+                  value: index,
+                  child: Wrap(
+                    children: [
+                      Padding(
+                        padding: const EdgeInsets.only(left: 10),
+                        child: Icon(
+                          index == colorSelected
+                              ? Icons.color_lens
+                              : Icons.color_lens_outlined,
+                          color: colorOptions[index],
+                        ),
+                      ),
+                      Padding(
+                          padding: const EdgeInsets.only(left: 20),
+                          child: Text(colorText[index]))
+                    ],
+                  ));
+            });
+          },
+          onSelected: handleColorSelect,
+        ),
+      ],
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      debugShowCheckedModeBanner: false,
+      title: 'Material 3',
+      themeMode: onLightMode ? ThemeMode.light : ThemeMode.dark,
+      theme: onLightMode ? lightTheme : darkTheme,
+      home: LayoutBuilder(builder: (context, constraints) {
+        if (constraints.maxWidth < 450) {
+          return Scaffold(
+            appBar: handleAppBar(),
+            body: Row(children: <Widget>[
+              handleScreenChange(screenIndex, false),
+            ]),
+            bottomNavigationBar: NavigationBars(
+              onSelectItem: onNavigationSelect,
+              selectedIndex: screenIndex,
+              isExampleBar: false,
+            ),
+          );
+        } else {
+          return Scaffold(
+            appBar: handleAppBar(),
+            body: SafeArea(
+              bottom: false,
+              top: false,
+              child: Row(children: <Widget>[
+                Padding(
+                    padding: const EdgeInsets.symmetric(horizontal: 5),
+                    child: NavigationRailSection(
+                        onSelectItem: onNavigationSelect,
+                        selectedIndex: screenIndex)),
+                const VerticalDivider(thickness: 1, width: 1),
+                handleScreenChange(screenIndex, true),
+              ]),
+            ),
+          );
+        }
+      }),
+    );
+  }
+}
+
+class ComponentScreen extends StatelessWidget {
```

Comment:
This might be good to pull out into its own file like the other screens. That way the main.dart is just the app skeleton.